### DISCUSSION TITLE
When running queries in Greenplum the Discover Owner Entity task takes 30+ seconds to complete

### DISCUSSION THREAD

**Discussion by charlesdmoorejr** [dbeaver/dbeaver#14529] [Q&A] [answered]
#### System information: 
- Operating system (distribution) and version - Windows 10
- DBeaver version - 21.0.3
- Additional extensions

#### Connection specification:
- Database name and version - Greenplum version 5.28.6
- Driver name - PostgreSQL JDBC Driver
- Do you use tunnels or proxies (SSH, SOCKS, etc)? - No

#### Describe the problem you're observing:

When running queries on any of our tables DBeaver first runs a couple of queries to collect metadata about objects being queried known as "Discover Owner Entity".  The query pasted below is run by DBeaver in the background and takes upwards of 30 seconds to complete we think in part due to the JOIN condition for table partitions. We have quite a few schemas/partitions which could affect this query's performance. 

I haven't been able to find an option in the Greenplum connection to disable the Discover Owner Entity task. Using a Postgres connection though pointing at our Greenplum database it doesn't face this issue (as DBeaver Postgres connection shows partitions as separate tables). 

Is there an option to disable the Discover Owner Entity task? I see in other connection types it's possible. 

```
SELECT c.oid,d.description,p.partitiontablename,c.*,
CASE WHEN x.urilocation IS NOT NULL THEN array_to_string(x.urilocation, ',') ELSE '' END AS urilocation,
CASE WHEN x.command IS NOT NULL THEN x.command ELSE '' END AS command,
x.fmttype, x.fmtopts,
coalesce(x.rejectlimit, 0) AS rejectlimit,
coalesce(x.rejectlimittype, '') AS rejectlimittype,
array_to_string(x.execlocation, ',') AS execlocation,
pg_encoding_to_char(x.encoding) AS encoding,
x.writable,
case when x.fmterrtbl is not NULL then true else false end as "is_logging_errors",
case when c.relstorage = 'x' then true else false end as "is_ext_table",
case when (ns.nspname !~ '^pg_toast' and ns.nspname like 'pg_temp%') then true else false end as "is_temp_table"
FROM pg_catalog.pg_class c
INNER JOIN pg_catalog.pg_namespace ns
    on ns.oid = c.relnamespace
LEFT OUTER JOIN pg_catalog.pg_description d
    ON d.objoid=c.oid AND d.objsubid=0
LEFT OUTER JOIN pg_catalog.pg_exttable x
    on x.reloid = c.oid
LEFT OUTER JOIN pg_catalog.pg_partitions p
    on c.relname = p.partitiontablename and ns.nspname = p.schemaname
WHERE c.relnamespace= $1 AND c.relkind not in ('i','c')  AND relname=$2
```

#### Steps to reproduce, if exist:

Running a SELECT statement against a user created schema/table. Must have a large number of partitions/schemas. 

#### Include any warning/errors/backtraces from the logs
No warnings/errors/etc.

**Comment by uslss** [comment, marked as answer]
Thank you for the report. As a workaround you can disable Read table metadata and Read Table references options in Preferences
![image](https://user-images.githubusercontent.com/31996417/123966721-58195180-d9be-11eb-9316-e922c05cfef2.png)

**Comment by charlesdmoorejr** [comment]
Excellent, this seems to significantly reduce the time on those background tasks.
